HerPublicWebsite/Middleware/SecurityHeadersMiddleware.cs

```diff
@@ -34,4 +43,12 @@ public Task Invoke(HttpContext context)
 
         return next(context);
     } 
-}
\ No newline at end of file
+}
+
+public static class SecurityContextExtensions
+{
+    public static HtmlString GetScriptNonce(this HttpContext context)
+    {
+        return new HtmlString((string) context.Items[SecurityHeadersMiddleware.NonceContextKey] ?? "");
+    }
```

Comment:
I wasn't sure a few aspects of this, so happy to move if you prefer
- Should this be an extension method? We could also use `Context.Items[SecurityHeadersMiddleware.NonceContextKey]` in the views, it's just a bit ugly
- Should it be an extension method on the `HttpContext`, `HtmlHelper` or something else? The context seemed like the most direct
- Should this be in a separate file? I like having it close to the rest of the CSP context